perun-core/src/main/java/cz/metacentrum/perun/core/entry/AttributesManagerEntry.java

```diff
@@ -1815,12 +1918,53 @@ public void deleteAttribute(PerunSession sess, AttributeDefinition attributeDefi
 				if(!AuthzResolver.isAuthorizedForAttribute(sess, ActionType.READ, attrNext, member, null)) attrIter.remove();
 				else attrNext.setWritable(AuthzResolver.isAuthorizedForAttribute(sess, ActionType.WRITE, attrNext, member, null));
 			} else if (getAttributesManagerBl().isFromNamespace(sess, attrNext, NS_USER_ATTR)) {
-				User u = getPerunBl().getUsersManagerBl().getUserByMember(sess, member);
+				if (u==null) u = getPerunBl().getUsersManagerBl().getUserByMember(sess, member);
 				if(!AuthzResolver.isAuthorizedForAttribute(sess, ActionType.READ, attrNext, u, null)) attrIter.remove();
 				else attrNext.setWritable(AuthzResolver.isAuthorizedForAttribute(sess, ActionType.WRITE, attrNext, u, null));
 			} else if (getAttributesManagerBl().isFromNamespace(sess, attrNext, NS_USER_FACILITY_ATTR)){
-				User u = getPerunBl().getUsersManagerBl().getUserByMember(sess, member);
-				Facility f = getPerunBl().getResourcesManagerBl().getFacility(sess, resource);
+				if (u==null) u = getPerunBl().getUsersManagerBl().getUserByMember(sess, member);
+				if (f==null) f = getPerunBl().getResourcesManagerBl().getFacility(sess, resource);
+				if(!AuthzResolver.isAuthorizedForAttribute(sess, ActionType.READ, attrNext, u, f)) attrIter.remove();
+				else attrNext.setWritable(AuthzResolver.isAuthorizedForAttribute(sess, ActionType.WRITE, attrNext, u, f));
+			} else {
+				throw new ConsistencyErrorException("There is some attribute which is not type of any possible choice.");
+			}
+		}
+		return attributes;
+	}
+
+	public List<Attribute> getRequiredAttributes(PerunSession sess, Service service, Resource resource, Group group, Member member, boolean workWithUserAttributes) throws PrivilegeException, InternalErrorException, ResourceNotExistsException, ServiceNotExistsException, MemberNotExistsException, WrongAttributeAssignmentException, GroupNotExistsException {
+		Utils.checkPerunSession(sess);
+		getPerunBl().getServicesManagerBl().checkServiceExists(sess, service);
+		getPerunBl().getResourcesManagerBl().checkResourceExists(sess, resource);
+		getPerunBl().getGroupsManagerBl().checkGroupExists(sess, group);
+		getPerunBl().getMembersManagerBl().checkMemberExists(sess, member);
+
+		List<Attribute> attributes = getAttributesManagerBl().getRequiredAttributes(sess, service, resource, group, member, workWithUserAttributes);
+
+		User u = null;
+		Facility f = null;
```

Comment:
Please use full name of variable "user" instead of just "u" when you are already working with this object. And also please call it from this place always. It is just one calling of method getUserByMember and one calling of method getFacility (by Resource) and it will be more human readable.